dcrjson/parse_test.go

```diff
@@ -46,3 +49,133 @@ func TestDecodeConcatenatedHashes(t *testing.T) {
 		}
 	}
 }
+
+func TestEncodeConcatenatedVoteBits(t *testing.T) {
+	testVbs := []stake.VoteBits{
+		stake.VoteBits{Bits: 0, ExtendedBits: []byte{}},
+		stake.VoteBits{Bits: 0, ExtendedBits: []byte{0x00}},
+		stake.VoteBits{Bits: 0x1223, ExtendedBits: []byte{0x01, 0x02, 0x03, 0x04}},
+		stake.VoteBits{Bits: 0xaaaa, ExtendedBits: []byte{0x01, 0x02, 0x03, 0x04, 0x05}},
+	}
+	encodedResults := make([][]byte, len(testVbs))
+	var err error
+	for i, vbs := range testVbs {
+		encodedResults[i], err =
+			dcrjson.EncodeConcatenatedVoteBits(vbs.Bits, vbs.ExtendedBits)
+		if err != nil {
+			t.Fatalf("Encode failed: %v", err)
+		}
+	}
+
+	expectedEncoded := [][]byte{
+		[]byte{0x02, 0x00, 0x00},
+		[]byte{0x03, 0x00, 0x00, 0x00},
+		[]byte{0x06, 0x23, 0x12, 0x01, 0x02, 0x03, 0x04},
+		[]byte{0x07, 0xaa, 0xaa, 0x01, 0x02, 0x03, 0x04, 0x05},
+	}
+
+	expectedEncodedBytes := make([]byte, 0)
+	for i, expected := range expectedEncoded {
+		if !bytes.Equal(expected, encodedResults[i]) {
+			t.Fatalf("Encoded votebits %d `%x` does not match expected `%x`",
+				i, encodedResults[i], expected)
+		}
+
+		expectedEncodedBytes = append(expectedEncodedBytes, expected...)
+	}
+
+	// Double check that decoding produces the expected result, too.
```

Comment:
This test should be moved to a new test file (currently the test func name lies).  The test vectors can be made package globals so they can be referenced by each test.